lib/js/components/Modals/Modal/Modal.vue

```diff
@@ -0,0 +1,417 @@
+<template>
+	<div
+		class="modal"
+		:class="{
+			'-danger': danger,
+		}"
+		@click.stop="$emit('close-modal')"
+	>
+		<div class="modal__wrapper" :class="{ '-small': size === MODAL_SIZES.SMALL }">
+			<div
+				v-if="headerImage"
+				class="modal__image"
+				:style="{ backgroundImage: 'url(' + headerImage + ')' }"
+			></div>
+			<div class="modal__content" :class="{ '-centered': contentCentered }">
+				<div class="modal__header">
+					<wnl-icon-button
+						touchable
+						:icon="ICONS.FA_XMARK"
+						class="modal__close"
+						:size="ICON_SIZES.SMALL"
+						:elevation="BUTTON_ELEVATIONS.X_SMALL"
+						:color="ICON_BUTTON_COLORS.NEUTRAL_WEAK"
+						@click.native.stop="$emit('close-modal')"
+					/>
+					<feature-icon
+						v-if="headerFeatureIcon"
+						class="modal__headerFeatureIcon"
+						:color="calcHeaderFeatureIconColor"
+						:icon="headerFeatureIcon"
+						:size="FEATURE_ICON_SIZES.X_LARGE"
+						:double-background="true"
+					/>
+					<h3
+						class="modal__headerTitle"
+						:class="{ '-small': headerTitleSize === MODAL_HEADER_TITLE_SIZES.SMALL }"
+						>{{ headerTitle }}</h3
+					>
+					<h6 v-if="headerSubtitle" class="modal__headerSubtitle">{{
+						headerSubtitle
+					}}</h6>
+				</div>
+				<div v-if="$slots.default" class="modal__textContent">
+					<slot />
+				</div>
+				<div class="modal__footer" :class="{ '-singleColumn': calcSingleColumn }">
+					<div
+						v-if="footerTertiaryButtonText || footerCheckboxText"
+						class="modal__footerColumn --ctaSecondary"
+					>
+						<div v-if="footerCheckboxText" ref="button" class="modal__checkbox">
```

Comment:
what is the benefit of using `ref="button"` in this component?

Reply:
No benefit, copy paste mistake.

Reply:
I still see `ref` defined for other buttons

Reply:
Fixed.